lms/services/async_oauth_http.py

```diff
@@ -44,6 +44,10 @@ def request(
 
 async def _async_request(aio_session, method, url, **kwargs):
     async with aio_session.request(method, url, **kwargs) as response:
+        # Calling `.text()` here caches the result in `response` but is still behind a coroutine.
+        # We assign it to another response attribute for it to be
+        # available in a sync context without needing to start coroutine.
+        response.sync_text = await response.text()
```

Comment:
The alternative to this would be to to so something like 

https://github.com/hypothesis/lms/blob/38a08c221b0e40d669d3d4db10cb8e6c22667749/lms/services/async_oauth_http.py#L40

In the exception class. Having a corutine to get the text and run it in another thread with async.run.

I think the method ins this PR, get the body when we are already in a async context, might be useful if we ever needed the responses text in for example the BB service while having all the async code only in `async_oauth_service`.

For now it works well as we already `getattr(response...` in the exception for other properties.

Reply:
:man_shrugging: Whatever works I guess :)